PythonPonies/bangazon-storefront
# Customer Can Checkout & Complete Order

# Description
This pull request proposes merging code that will allow a customer to complete an order by selecting a payment type to use & confirming payment. 

Also included are tests for the Checkout view. 

## Number of Fixes
0 fixes. New code. 

## Related Ticket(s)
#10


## Problem to Solve
Until this point, the customer has had no way of completing an order that they have created. We would like to enable the customer to choose a payment type from their existing payment types & confirm order completion. 

## Expected Behavior
The customer should now be able to functionally complete each step in the "shopping" process. Register, login, add payment types to account, add products to cart, view order, and complete order, or, "checkout".

## Steps to Test Solution

1. git fetch
2. git checkout swh-completeorder
3. rm db.sqlite3
4. python manage.py migrate
5. python manage.py test bangazon_storefront
6. python manage.py createsuperuser
7. python manage.py runserver
8. direct browser to localhost:8000/admin
9. create product types + products
10. direct browser to localhost:8000/register
11. walk thru the steps of registering a customer, adding payment types, adding products to cart.
12. go to localhost:8000/checkout
13. complete order.
14. you will be able to see the appropriate order row change in the database when order is completed. the payment_complete column will now be "1", and the payment_type_id will now be the pk of the chosen payment type instead of null. 

## Testing

[x] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[x] I certify that all existing tests pass (except one that is unrelated to this PR.)

## Documentation
[x] There is new documentation in this pull request that must be reviewed..
[x] I added documentation for any new classes/methods

## Deploy Notes
N/A
